core/services/relay/evm/chain_writer.go

```diff
@@ -0,0 +1,118 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/google/uuid"
+	commonservices "github.com/smartcontractkit/chainlink-common/pkg/services"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+)
+
+type ChainWriter interface {
```

Comment:
Why isn't the interface definition? pulled from common

Reply:
It will be. If you see the description of the PR, this will be removed as soon as the PR in common is merged. I added the interface here temporarily so the build didn't break.

Reply:
ah my bad, you can reference the PR in common in this PR before it is merged